### PR TITLE
New Makeshift weaponry: Ross Rifle and Huot Conversion!

### PR DESCRIPTION
**Pull request by PapaSejd**
## About The Pull Request

Adds 2 new weapons and a design for the autolathe.

- Ross Rifle. Once a standard rifle in the Canadian military in WW1, this rifle is now craftable in the recipe menu, and requires roughly the same materials as its other makeshift brothers. Does 25 damage and a flat 10% chance to jam. Holds 5 .303 bullets.

image

- Huot Conversion. This conversion recipe can convert the Ross rifle into another variant of it, the Huot Automatic. Designed in 1916, this automatic rifle holds 25 shots in a drum mag, with a flat chance of 5% to jam. Requires a firing pin (new firing system) and more materials to create than its other brothers. It has a firing rate of 2 (less than a stetchkin).

image

- Adds the .303 bullet to hacked Autolathe designs.

## Why It's Good For The Game

To complete the trifecta of weapon types, rifles are now added for players to craft. Now assistants can find their niche when trying to kill all of security!

## Changelog
:cl:
add: added rifles, bullets and more
/:cl:

### CODE REVIEW

**Comment by thelaughingbomb** [comment]
Nice

File: austation/code/datums/components/crafting/recipes.dm [hunk 1]
```diff
@@ -10,3 +10,31 @@
...
+				/obj/item/weaponcrafting/stock = 1,
+				/obj/item/stack/packageWrap = 10,
+				/obj/item/stack/rods = 10,
+				/obj/item/stack/sheet/iron = 5)
+	tools = list(TOOL_SCREWDRIVER)
+	time = 200
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+	/datum/crafting_recipe/huotconvert
```
Comment (thelaughingbomb) [inline austation/code/datums/components/crafting/recipes.dm:28]: I think travis is being pissy about the space at the start of the line

File: austation/code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	spread = 7
+	w_class = WEIGHT_CLASS_HUGE
+	bolt_type = BOLT_TYPE_STANDARD
+
+/obj/item/gun/ballistic/automatic/huot/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "") //creds to thelaughingbomb, very nice way to make jams
+	if(prob(5)) 
+		playsound(user, fire_sound, fire_sound_volume, vary_fire_sound)
+		user.visible_message("<span class='userdanger'>The [src] has misfired and requires to be rechambered!</span>")
+		qdel(chambered)
+		chambered = null
```
Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/ballistic/automatic.dm:21]: You can instantly unjam by pressing z, maybe add an action timer so it actually does something

File: austation/code/modules/projectiles/guns/ballistic/rifle.dm [hunk 3]
```diff
@@ -0,0 +1,21 @@
+///////////////////////
+//     ROSS RIFLE    //
+///////////////////////
+
+/obj/item/gun/ballistic/rifle/ross //dont craft this unless you want sec to laugh at you
+	name = "Improvised Ross Rifle"
+	desc = "Already a terrible rifle, this improvised Ross rifle should be able to take care of security officers, maybe."
```
Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/ballistic/rifle.dm:7]: Maybe something more along the lines of explaining the weapon

File: austation/code/modules/projectiles/guns/ballistic/automatic.dm [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	name = "Huot-Ross Conversion Rifle"
+	desc = "Converted from the poor-quality Ross rifle, this robust automatic rifle from the bygone eras of WW1 should prove a formidable foe!"
+	icon = 'austation/icons/obj/guns.dmi'
+	icon_state = "huot"
+	slot_flags = 0
+	mag_type = /obj/item/ammo_box/magazine/huot
+	fire_sound = 'sound/weapons/rifleshot.ogg'
+	rack_sound = 'sound/weapons/chunkyrack.ogg'
+	can_suppress = FALSE
+	fire_rate = 3
```
Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/ballistic/automatic.dm:11]: 90 DPS for a makeshift weapon, pls no

File: austation/code/datums/components/crafting/recipes.dm [hunk 5]
```diff
@@ -1,3 +1,44 @@
...
+
+/datum/crafting_recipe/huotconvert
+	name = "Huot Conversion"
+	result = /obj/item/gun/ballistic/automatic/huot
+	reqs = list(/obj/item/gun/ballistic/rifle/ross,
+				/obj/item/pipe = 1,
+				/obj/item/stack/packageWrap = 10,
+				/obj/item/stack/rods = 10,
+				/obj/item/stack/sheet/iron = 20,
+				/obj/item/firing_pin/ = 1)
```
Comment (TheFakeElon) [inline austation/code/datums/components/crafting/recipes.dm:36]: That’s pretty cheap considering how strong this weapon is, rivalling some syndicate and security exclusive weapons.

File: austation/code/modules/projectiles/guns/ballistic/automatic.dm (hunk 2 above)
Comment (TheFakeElon) [reply austation/code/modules/projectiles/guns/ballistic/automatic.dm:21]: And yes, I know this is how the makeshift pistol does it but it’s nowhere near as strong as this.

**Comment by MCterra10** [comment]
testmerging

**Comment by PapaSejd** [comment]
edited to include #2171 changes

**Review by MCterra10 — CHANGES_REQUESTED** [review]
can you knock the damage down a bit? 75dps with only a 5% chance to jam and a 25 round mag is pretty ridiculous

**Comment by PapaSejd** [comment]
Aight ill drop it down to 15, it will render the ross essentially useless but I guess its only meant to be a stepping stone for the huot

**Comment by MCterra10** [comment]
Conflicts my guy

**Comment by PapaSejd** [comment]
Im gonna have to close this for now and reopen another PR cause the branch was deleted lol

**Comment by MCterra10** [comment]
f

**Pull request closed without merging**